src/main/java/net/pms/dlna/RootFolder.java

```diff
@@ -1514,6 +1514,9 @@ public void notify(String filename, String event, FileWatcher.Watch watch, boole
 							} else {
 								LOGGER.trace("Folder {} is empty", filename);
 							}
+						} else if ("ENTRY_DELETE".equals(event)) {
```

Comment:
@SubJunk this doesn't work for me. It was also my idea to fix it in that way but there is a condition https://github.com/UniversalMediaServer/UniversalMediaServer/blob/5f8da71ce15a223fb90fb145166baf197ddce88d/src/main/java/net/pms/dlna/RootFolder.java#L1501 which is always false for folders.

Reply:
Oh that's weird... I tested on Windows 7 so that must be why we got different results. Which version of Java?

Reply:
We can do a hack around it anyway by detecting whether the string is a folder, and even if that detection gets it wrong (like if it is a file with no extension) it won't cause any big problems, I think it would just delete the file instead, and if it doesn't the file will be deleted during the cleanup step of the next scan

Reply:
@SubJunk I don't know if it works for WIN7 but it is logical that the code at line 301 doesn't work for deleted folders because the `WatchEvent` is raised after the OS deletes the folder and in that time no longer exists on the drive. The `Files.isDirectory()` is checking if the folder exists and if not returns `false` like it is not folder.